### PR TITLE
When video is playing, save position intermittently

### PR DESCRIPTION
**Pull request by david0178418**
## What is the current behavior?
Currently, the position of a video is saved when pausing, seeking, or navigating to a url that is handled by the router. However, if you close the tab or navigate away from the site with a hard link, the play position is lost.

## What is the new behavior?
This patch will intermittently save while the video is playing to better prevent losing video progress.  An arbitrary interval of 15 seconds was chose, but this can be adjusted if there is some more appropriate interval.

## PR Checklist

<details><summary>Toggle...</summary>

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

</details>

### CODE REVIEW

**Comment by tzarebczan** [comment]
Thanks David! Not sure how I feel about 1500 seconds - I'm sure the performance implications are low, but we can probably get away with 5-10 seconds. What do you think?

**Comment by david0178418** [comment]
Ah, my mistake.  I'd upped it to 15 seconds while getting the PR set, but forgot to push that commit.  Just pushed it now.  Does 15 seconds work, or would you like me to lower it 5 or 10?

**Comment by tzarebczan** [comment]
15 seconds should be perfect, thanks again!

File: ui/component/viewers/videoViewer/view.jsx [hunk 1]
```diff
@@ -27,9 +27,11 @@ import { getAllIds } from 'util/buildHomepage';
 import type { HomepageCat } from 'util/buildHomepage';
 import debounce from 'util/debounce';
 import { formatLbryUrlForWeb, generateListSearchUrlParams } from 'util/url';
+import useInterval from 'effects/use-interval';
 
 // const PLAY_TIMEOUT_ERROR = 'play_timeout_error';
 // const PLAY_TIMEOUT_LIMIT = 2000;
+const PLAY_POSITION_SAVE_INTERVAL = 15000;
```
Comment (infinite-persistence) [inline ui/component/viewers/videoViewer/view.jsx:34]: Minor:  maybe rename to `PLAY_POSITION_SAVE_INTERVAL_MS`, although it is obvious.

**Review by infinite-persistence — COMMENTED** [review]
Looks goods.  Would be nice if the interval isn't running when paused or ended, but that's minor.

A few more minor tweaks and good to go.  Thanks!

**Comment by david0178418** [comment]
> Would be nice if the interval isn't running when paused or ended

Yeah, I was sorta torn on that one.  To do that, the code is just a pinch more cryptic.  Weighing the complexity against the ever-so-slight increase in code clarity with the overhead of an extra function call every 15 seconds when idle, I tipped to the latter.

I went ahead and implemented the changes to only run when playing in the most recent commit here: https://github.com/OdyseeTeam/odysee-frontend/pull/1013/commits/5ce6ab9689c1610f24869c7347b823cc2023b767

It's not a huge difference either way, but let me know what you think.  I can keep or revert that commit. And, of course, I can implement any other suggestion on that front.

**Comment by infinite-persistence** [comment]
Yeah, I thought the previous one was a good tradeoff for clarity, less complex and one less state.  Sorry, should have made it clear in my previous sentence that it was ok, knowing the tradeoffs.  But at least we have a reference on how to do that if we really need it.

Let's revert to the simpler ref version?  Thanks!

**Comment by david0178418** [comment]
No worries. Just rolled that one back.

**Pull request merged**